packages/swingset-liveslots/src/collectionManager.js

```diff
@@ -48,6 +49,58 @@ function throwNotDurable(value, slotIndex, serializedValue) {
   );
 }
 
+function prefixc(collectionID, dbEntryKey) {
+  return `vc.${collectionID}.${dbEntryKey}`;
+}
+
+/**
+ * @typedef {object} SchemaCacheValue
+ * @property {Pattern} keyShape
+ * @property {Pattern} valueShape
+ * @property {string} label
+ * @property {object} schemataCD
+ */
+
+/*
+ * Build a cache that holds the schema for each collection.
+ *
+ * The cache maps colectionID to { keyShape, valueShape, label
+ * }. These are initialized when the collection is first constructed,
+ * and never modified afterwards. The values live in the vatstore,
+ * inside two keys, one for the [keyShape, valueShape] schemata,
+ * another for the label.
+ */
+function makeSchemaCache(syscall, unserialize) {
+  /** @type {(collectionID: string) => SchemaCacheValue} */
+  const readBacking = collectionID => {
+    // this is only called once per crank
+    const schemataKey = prefixc(collectionID, '|schemata');
+    const schemataValue = syscall.vatstoreGet(schemataKey);
+    const schemataCD = JSON.parse(schemataValue);
+    const { keyShape, valueShape } = unserialize(schemataCD);
+    const labelKey = prefixc(collectionID, '|label');
+    const label = syscall.vatstoreGet(labelKey);
+    return harden({ keyShape, valueShape, label, schemataCD });
+  };
+  /** @type {(collectionID: string, value: SchemaCacheValue) => void } */
+  const writeBacking = (collectionID, value) => {
+    const { label, schemataCD } = value;
+    const schemataKey = prefixc(collectionID, '|schemata');
+    const schemataValue = JSON.stringify(schemataCD);
```

Comment:
This is preparation for #7321  (PR #7334), which needs to know about any slots (vrefs) in the serialized keyShape/valueShape. When the collection is deleted, rather than re-serialize those values (to learn the vrefs, so we can feed them to `vrm.removeReachableVref`, to decrement their refcounts), we keep the original serialized forms around. The memory overhead of retaining the serialized capdata is pretty small, and it didn't occur to me to trust that serialization would return the same values as the original.. felt more appropriate to retain the original instead.

The only time we write to the `schemaCache` is when a collection is first created, and at that time we also have the serialized capdata available (again, as anticipation for #7334, which needs to *increment* refcounts for everything therein).